src/valkey-cli.c

```diff
@@ -9010,15 +9020,30 @@ static void getKeyFreqs(redisReply *keys, unsigned long long *freqs) {
     }
 }
 
-#define HOTKEYS_SAMPLE 16
 static void findHotKeys(void) {
     redisReply *keys, *reply;
-    unsigned long long counters[HOTKEYS_SAMPLE] = {0};
-    sds hotkeys[HOTKEYS_SAMPLE] = {NULL};
+    unsigned long long *counters = NULL;
+    sds *hotkeys = NULL;
     unsigned long long sampled = 0, total_keys, *freqs = NULL, it = 0, scan_loops = 0;
     unsigned int arrsize = 0, i, k;
     double pct;
 
+    counters = zrealloc(counters, sizeof(unsigned long long) * config.hotkeys_count);
+    if (!counters) {
+        fprintf(stderr, "Memory allocated failed!\n");
+        exit(1);
+    }
+    hotkeys = zrealloc(hotkeys, sizeof(sds) * config.hotkeys_count);
+    if (!hotkeys) {
+        fprintf(stderr, "Memory allocated failed!\n");
+        exit(1);
+    }
```

Comment:
`zrealloc()` never returns NULL. If it fails to allocate, it calls the oom handler which exits the program:

```
static void zmalloc_default_oom(size_t size) {
    fprintf(stderr, "zmalloc: Out of memory trying to allocate %zu bytes\n", size);
    fflush(stderr);
    abort();
}
```

So we can delete the error handling here.

```suggestion
    counters = zrealloc(counters, sizeof(unsigned long long) * config.hotkeys_count);
    hotkeys = zrealloc(hotkeys, sizeof(sds) * config.hotkeys_count);
```

Reply:
Thanks, I learn it.